pagerduty/resource_pagerduty_custom_field_option.go

```diff
@@ -0,0 +1,182 @@
+package pagerduty
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/heimweh/go-pagerduty/pagerduty"
+)
+
+func resourcePagerDutyCustomFieldOption() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   resourcePagerDutyCustomFieldOptionRead,
+		UpdateContext: resourcePagerDutyCustomFieldOptionUpdate,
+		DeleteContext: resourcePagerDutyCustomFieldOptionDelete,
+		CreateContext: resourcePagerDutyCustomFieldOptionCreate,
+		CustomizeDiff: validateCustomFieldOptionValue,
+		Schema: map[string]*schema.Schema{
+			"field": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"datatype": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validateCustomFieldDataType(),
+			},
+			"value": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func validateCustomFieldOptionValue(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
```

Comment:
Could you please comment on the code what is the scenario needing this customizing diff logic?

Reply:
I put the comment in the schema.Resource definition at the top of this and in `resource_custom_field_schema_field_configuration` (which does something similar)